databus-client2/src/main/java/com/bazaarvoice/emodb/databus/client2/client/DatabusClient.java

```diff
@@ -0,0 +1,553 @@
+import com.bazaarvoice.emodb.databus.api.UnknownSubscriptionException;
+import com.bazaarvoice.emodb.databus.api.UnknownMoveException;
+import com.bazaarvoice.emodb.databus.api.UnknownReplayException;
+import com.bazaarvoice.emodb.databus.api.PollResult;
+import com.bazaarvoice.emodb.databus.api.ReplaySubscriptionStatus;
+import com.bazaarvoice.emodb.databus.api.MoveSubscriptionStatus;
+import com.bazaarvoice.emodb.databus.api.Event;
+import com.bazaarvoice.emodb.databus.client2.discovery.DatabusDiscovery;
+import com.bazaarvoice.emodb.sor.condition.Condition;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriBuilder;
+import java.io.Closeable;
+import java.io.IOException;
+import java.net.URI;
+import java.text.SimpleDateFormat;
+import java.time.Duration;
+import java.util.Collection;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.TimeZone;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+
+/**
+ * Thin client for accessing EmoDB Databus.
+ */
+public class DatabusClient implements Databus, Closeable {
+
+    private final static Logger _log = LoggerFactory.getLogger(com.bazaarvoice.emodb.databus.client2.client.DatabusClient.class);
+
+    private static final String POLL_DATABUS_EMPTY_HEADER = "X-BV-Databus-Empty";
+    private static final MediaType JSON_CONDITION_MEDIA_TYPE = new MediaType("application", "x.json-condition");
+
+    private final DatabusDiscovery _databusDiscovery;
+    private final EmoClient _client;
+    private final String _apiKey;
+    private final boolean _partitionSafe;
+
+    public DatabusClient(DatabusDiscovery databusDiscovery, EmoClient client, String apiKey) {
+        _databusDiscovery = databusDiscovery;
+        _client = client;
+        _apiKey = apiKey;
+        _partitionSafe = false;
+    }
+
+    @Override
+    public Iterator<Subscription> listSubscriptions(@Nullable String fromSubscriptionExclusive, long limit) {
+        checkArgument(limit > 0, "Limit must be >0");
+        Response response = null;
+        try {
+            URI uri = UriBuilder.fromUri(_databusDiscovery.getBaseUri())
+                    .path("bus")
+                    .path("1")
+                    .queryParam("from", optional(fromSubscriptionExclusive))
+                    .queryParam("limit", limit)
+                    .build();
+
+            return _client.resource(uri)
+                    .accept(MediaType.APPLICATION_JSON_TYPE)
+                    .header(ApiKeyRequest.AUTHENTICATION_HEADER, _apiKey)
+                    .get(new TypeReference<Iterator<Subscription>>() {});
+        }catch (Exception e) {
```

Comment:
here and below, reformat code - https://www.jetbrains.com/help/idea/reformat-and-rearrange-code.html#reformat_file